packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -474,15 +480,33 @@ export async function toAbsoluteGlob(
     pre = '!'
     glob = glob.slice(1)
   }
-
-  const dir = importer ? dirname(importer) : root
+  root = globSafePath(root)
+  const dir = importer ? globSafePath(dirname(importer)) : root
   if (glob.startsWith('/')) return pre + posix.join(root, glob.slice(1))
   if (glob.startsWith('./')) return pre + posix.join(dir, glob.slice(2))
   if (glob.startsWith('../')) return pre + posix.join(dir, glob)
   if (glob.startsWith('**')) return pre + glob
 
   const resolved = normalizePath((await resolveId(glob, importer)) || glob)
-  if (isAbsolute(resolved)) return pre + resolved
+  if (isAbsolute(resolved)) {
+    // we have to escape special glob characters in the resolved path, but keep the user specified globby suffix
+    // walk back both strings until a character difference is found
+    // then slice up the resolved path at that pos and escape the first part
+    let similarEndLength = 0
+    while (
+      resolved.charAt(resolved.length - 1 - similarEndLength) ===
+        glob.charAt(glob.length - 1 - similarEndLength) &&
+      similarEndLength < glob.length &&
+      similarEndLength < resolved.length
+    ) {
+      similarEndLength += 1
+    }
+    const staticPartEnd = resolved.length - similarEndLength
+    const staticPart = resolved.slice(0, staticPartEnd)
+    const dynamicPart = resolved.slice(staticPartEnd)
+    const safeResolved = globSafePath(staticPart) + dynamicPart
```

Comment:
The logic and test LGTM. I think it'd be nice if we can extract this part as a function.

Reply:
refactored. the extra function call for nth last is maybe a bit over the top, but makes the implementation a lot nicer to read

Reply:
I was actually thinking of something like `splitByMatchingEnding()` but this works too 😅